app/javascript/lib/validation/validate-file-content.js

```diff
@@ -276,99 +279,138 @@ function validateClusterCoordinates(headers) {
-  let friendlyDelimiter = 'tab'
-  if (delimiter === ',') {
-    friendlyDelimiter = 'comma'
-  } else if (delimiter === ' ') {
-    friendlyDelimiter = 'space'
-  }
+  let issues = validateCapFormat(parsedHeaders, delimiter)
+  issues = issues.concat(validateNoMetadataCoordinates(parsedHeaders))
+  // add other header validations here
 
-  const numRows = table.length
-  const numColumns = table[0].length
+  const dataObj = {} // object to track multi-line validation concerns
+  await chunker.iterateLines((line, lineNum, isLastLine) => {
+    const parsedLine = parseLine(line, delimiter)
+    issues = issues.concat(validateUniqueCellNamesWithinFile(parsedLine, isLastLine, dataObj))
+    // add other line-by-line validations here
+  })
+  return { issues, delimiter, numColumns: parsedHeaders[0].length }
+}
 
-  const defaultProps = {
-    fileType,
-    numRows,
-    numColumns,
-    numTableCells: numRows * numColumns,
-    delimiter: friendlyDelimiter,
-    fileName: file.name,
-    fileSize: file.size,
-    fileMimeType: file.type
-  }
+/** parse a cluster file, and return an array of issues, along with file parsing info */
+export async function parseClusterFile(chunker, mimeType) {
+  const { parsedHeaders, delimiter } = await getParsedHeaderLines(chunker, mimeType, 2)
 
-  if (errors.length === 0) {
-    return Object.assign({ status: 'success' }, defaultProps)
-  } else {
-    return Object.assign(defaultProps, {
-      status: 'failure',
-      summary,
-      numErrors: errors.length,
-      errors: errors.map(columns => columns[2]),
-      errorTypes: errors.map(columns => columns[1])
-    })
+  let issues = validateCapFormat(parsedHeaders, delimiter)
+  issues = issues.concat(validateClusterCoordinates(parsedHeaders))
+  // add other header validations here
+
+  const dataObj = {} // object to track multi-line validation concerns
+  await chunker.iterateLines((line, lineNum, isLastLine) => {
+    const parsedLine = parseLine(line, delimiter)
+    issues = issues.concat(validateUniqueCellNamesWithinFile(parsedLine, isLastLine, dataObj))
+    // add other line-by-line validations here
+  })
+
+  return { issues, delimiter, numColumns: parsedHeaders[0].length }
+}
+
+/** reads in the specified number of header lines, sniffs the delimiter, and returns the
+ * lines parsed by the sniffed delimiter
+ */
+export async function getParsedHeaderLines(chunker, mimeType, numHeaderLines=2) {
+  const headerLines = []
+  await chunker.iterateLines((line, lineNum, isLastLine) => {
+    headerLines.push(line)
+  }, 2)
+  if (headerLines.length < numHeaderLines || headerLines.some(hl => !hl)) {
+    throw new ParseException('format:cap:missing-header-lines', `Your file is missing newlines or is missing some of the required ${numHeaderLines} header lines`)
   }
+  const delimiter = sniffDelimiter(headerLines, mimeType)
+  const parsedHeaders = headerLines.map(l => parseLine(l, delimiter))
+  return { parsedHeaders, delimiter }
```

Comment:
This briefer name would provide a cleaner interface outside this function:

```suggestion
  const headers = headerLines.map(l => parseLine(l, delimiter))
  return { headers, delimiter }
```

Related to that goal, for consistency _within_ this function, I could also see renaming `headerLines` to `unparsedHeaderLines`.